pkg/controller/policyrecommendation_controller.go

```diff
@@ -18,59 +18,194 @@ package controller
 
 import (
 	"context"
-	"sigs.k8s.io/controller-runtime/pkg/controller"
-
+	"fmt"
+	"github.com/flipkart-incubator/ottoscalr/pkg/reco"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/tools/record"
 	ctrl "sigs.k8s.io/controller-runtime"
 	"sigs.k8s.io/controller-runtime/pkg/client"
-	"sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"time"
+
+	v1alpha1 "github.com/flipkart-incubator/ottoscalr/api/v1alpha1"
+)
+
+const (
+	POLICY_RECO_WORKFLOW_CTRL_NAME = "RecoWorkflowController"
+	EVENT_TYPE_NORMAL              = "Normal"
+	EVENT_TYPE_WARNING             = "Warning"
+)
 
-	ottoscaleriov1alpha1 "github.com/flipkart-incubator/ottoscalr/api/v1alpha1"
+var (
+	FALSE_BOOL = false
+	TRUE_BOOL  = true
 )
 
 // PolicyRecommendationReconciler reconciles a PolicyRecommendation object
 type PolicyRecommendationReconciler struct {
-	Client                  client.Client
+	client.Client
 	Scheme                  *runtime.Scheme
+	Recorder                record.EventRecorder
 	MaxConcurrentReconciles int
+	PolicyExpiryAge         time.Duration
+	RecoWorkflow            reco.RecommendationWorkflow
 }
 
 func NewPolicyRecommendationReconciler(client client.Client,
-	scheme *runtime.Scheme,
-	maxConcurrentReconciles int) *PolicyRecommendationReconciler {
+	scheme *runtime.Scheme, recorder record.EventRecorder,
+	maxConcurrentReconciles int, recommender reco.Recommender, policyIterators ...reco.PolicyIterator) *PolicyRecommendationReconciler {
+	recoWfBuilder := reco.NewRecommendationWorkflowBuilder().
+		WithRecommender(recommender)
+	for _, pi := range policyIterators {
+		recoWfBuilder = recoWfBuilder.WithPolicyIterator(pi.GetName(), pi)
```

Comment:
Can we only pass the policyOperator object in the builder fn? Passing the name seems unnecessary.

Reply:
This is helpful for printing logs with additional info such as the name.

Reply:
Will do. This signature was there before GetName() was part of the interface.